From 5826212d257bab080d396990a017306d0a133c49 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 22 Jun 2018 15:05:07 +0800
Subject: [PATCH 4101/5242] MLK-18685 usb: cdns3: gadget: balance spin lock if
 map request fails

commit  8e1113d7d931d6b5776187e06abde3c66ce6f3fa from
https://source.codeaurora.org/external/imx/linux-imx.git

spin_unlock_irqrestore is missing if usb_gadget_map_request_by_dev
fails.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index eefcddc..c81f974 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -1397,6 +1397,7 @@ static int usb_ss_gadget_ep0_queue(struct usb_ep *ep,
 	ret = usb_gadget_map_request_by_dev(usb_ss->sysdev, request,
 			usb_ss->ep0_data_dir);
 	if (ret) {
+		spin_unlock_irqrestore(&usb_ss->lock, flags);
 		dev_err(&usb_ss->dev, "failed to map request\n");
 		return -EINVAL;
 	}
-- 
1.7.9.5

